utils/s2n_random.c

```diff
@@ -542,38 +553,46 @@ static int s2n_rand_init_cb_impl(void)
     return S2N_SUCCESS;
 }
 
+bool s2n_supports_custom_rand()
+{
+#if !defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
+    return false;
+#endif
+
+    return !s2n_libcrypto_is_boringssl() && !s2n_libcrypto_is_libressl() && !s2n_libcrypto_is_awslc() && !s2n_is_in_fips_mode();
```

Comment:
The compiler doesn't complain about unreachable code when S2N_LIBCRYPTO_SUPPORTS_ENGINE is not defined? Usually this would be if/else

Reply:
Yea surprisingly it didn't. I'll refactor to if/else.